accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,290 @@
+
+The `salined` service must be able to attach to the Salt Master. At the moment this is done by connecting to Salt Master IPC sockets, which means `salined` needs to live in the same system/container than the Salt Master.
+
+Saline should be configurable to use IPC or TCP sockets. Switching Salt Master to use TCP sockets would enable Saline to run on a different container than the Salt Master.
+
+The Saline engine running on the Uyuni server, will provide data and metrics to feed Prometheus and Grafana instances and also a future UI with a live view of running actions. Eventually, this component should be integrated as part of the default Uyuni server stack.
+
+
+### Health Check Tool: An standalone tool you can run on an running Uyuni server or use in a disconnected setup (via supportconfig)
+In this case, this tool is mean not only to provide a picture of the current health status of a live running the Uyuni server, where the tool has access to it and can fetch data in real time, but it is meant to also help engineers and supporters to analyze and debug problems on disconnected setups, where the tool doesn't have access to an actual running Uyuni server, but only has a "supportconfig" as a source of data.
+
+In this regard, on disconnected setups, the tool cannot rely on existing Uyuni server components, neither existing Prometheus and Grafana instances, but it should be able to deploy its own instances to visualize the data coming from a "supportconfig".
+
+When the tool have access to an Uyuni server, then it can run "standalone" or via `mgradm`, allowing to resuse existing Prometheus and Grafana instances.
+
+- The Health Check Tool is not installed or running by default in a Uyuni server.
+- As multiple components (containers) are deployed, resources on a live Uyuni server might be affected.
+- It should be able to reuse existing Prometheus and Grafana instances, allowing integration with existing Monitoring stack.
+
+# How is the exporter going to access to uyuni-server container? For example, to query the database.
+As mentioned, the exporter needs to have access to Uyuni server container. It does need to query the DB and also execute some Salt runners jobs to gather metrics. If the "exporter" runs in a separated container than the Salt Master, then we can make the "exporter" container to run in the same network than the "uyuni-server", then TCP sockets are available to use.
+
+There are some Salt runners jobs that requires access to Salt Master Event publisher, currently exposed via IPC socket. This makes it tricky access these sockets from a different container. Some metrics might not be available until we move to "TCP" sockets for the Salt Master Event publisher in this case.
+
+Alternatively, we could make the "exporter" as part of the server-image, even if not running by default. Then it is up to the Healh Check Tool or Monitoring formula to start it. This would solve the current issue of not having access to the IPC sockets.
```

Comment:
```suggestion
Alternatively, we could make the "exporter" as part of the server-image, even if not running by default. Then it is up to the Health Check Tool or Monitoring formula to start it. This would solve the current issue of not having access to the IPC sockets.
```